src/colony/lua/preload.lua

```diff
@@ -73,24 +73,32 @@ if _G.COLONY_EMBED then
   end
 end
 
+local env = {}
 if not _G.COLONY_EMBED then
   -- This is temporary until we have proper compilation in C.
   colony._load = function (file)
     -- Compile JS script before running.
-    local status
+    assert(#_G.COLONY_COMPILER_PATH ~= 0, "COLONY_COMPILER_PATH is not defined")
+    local handle
     if jit == nil then
-      status = os.execute(_G.COLONY_COMPILER_PATH .. ' -m ' .. file .. ' > /tmp/colonyunique')
+      handle = io.popen(_G.COLONY_COMPILER_PATH .. ' -m "' .. file .. '"')
     else
-      status = os.execute(_G.COLONY_COMPILER_PATH .. ' -l ' .. file .. ' > /tmp/colonyunique')
+      handle = io.popen(_G.COLONY_COMPILER_PATH .. ' -l "' .. file .. '"')
     end
-    if status ~= 0 then
-      os.exit(status)
-    end
-    local file = io.open('/tmp/colonyunique', 'r')
-    local output = file:read('*all')
-    file:close()
+    local output = handle:read('*all')
+    handle:close()
     return output
   end
+
+  local handle = io.popen('env')
+  while true do
+    local line = handle:read('*line')
+    if line == nil then break end
+    for k, v in string.gmatch(line, "(.+)=(.*)") do
```

Comment:
This will break in the case of `export THING="a"$'\n'"b"` (which might be what we need for `TM_PROXY_CERT`)